categorize_notify/program.py

```diff
@@ -9,64 +9,106 @@
 SLACK_CHANNELS = ["demos", "engineering", "ui"]
+gmail = google.gmail_client("my_gmail").users()
+processed_message_ids = set()
+start_time = datetime.now(timezone.utc).timestamp()
 
 
 def on_http_get(event):
-    total_messages = None
     while True:
-        total_messages = _poll_inbox(total_messages)
-        time.sleep(10)
+        _poll_inbox()
+        time.sleep(float(POLL_INTERVAL))
+
+
+def _poll_inbox():
+    token = None
+    current_message_ids = set()
+
+    while True:
+        results = get_new_inbox_messages(token)
+
+        current_message_ids.update({msg["id"] for msg in results.get("messages", [])})
+
+        token = results.get("nextPageToken")
+        if not token:
+            break
+
+    new_message_ids = current_message_ids - processed_message_ids
+
+    for message_id in new_message_ids:
+        _process_email(message_id, start_time)
+
+    processed_message_ids.update(new_message_ids)
 
 
 @autokitteh.activity
-def _poll_inbox(prev_total: int):
-    gmail = google.gmail_client("my_gmail").users()
-    curr_total = gmail.getProfile(userId="me").execute()["messagesTotal"]
-    # Note: This is not meant to be a robust solution for handling email operations.
-    if prev_total and curr_total > prev_total:
-        new_email_count = curr_total - prev_total
-        message_ids = _get_latest_message_ids(gmail, new_email_count)
-        for message_id in message_ids:
-            _process_email(gmail, message_id)
+def get_new_inbox_messages(token):
+    query = "in:inbox -in:drafts"
+    if token:
+        gmail_list = gmail.messages().list(userId="me", q=query, pageToken=token)
+        results = gmail_list.execute()
+    else:
+        results = gmail.messages().list(userId="me", q=query).execute()
+    return results
 
-    return curr_total
 
+def _process_email(message_id: str, start_time: datetime):
+    message = get_message(message_id)
+    email_timestamp = float(message["internalDate"]) / 1000
+
+    if email_timestamp < start_time:
+        return
 
-def _process_email(gmail, message_id: str):
-    message = gmail.messages().get(userId="me", id=message_id).execute()
     email_content = _parse_email(message)
+
     if email_content:
         channel = _categorize_email(email_content)
+
         if channel:
             client = slack.slack_client("my_slack")
-            client.chat_postMessage(channel=channel, text=email_content)
+            text = email_content or "Email content not found."
+            client.chat_postMessage(channel=channel, text=text)
 
         # Add label to email
-        label_id = _get_label_id(gmail, channel) or _create_label(gmail, channel)
-        body = {"addLabelIds": [label_id]}
-        gmail.messages().modify(userId="me", id=message_id, body=body).execute()
+        label_id = _get_label_id(channel) or _create_label(channel)
+        if label_id:
+            body = {"addLabelIds": [label_id]}
+            apply_label_to_message(message_id, body)
+
+
+@autokitteh.activity
```

Comment:
Please recheck if the decorator is still needed after  https://github.com/autokitteh/autokitteh/pull/540

Reply:
It is only needed for the chatgpt call. Removed everywhere else.